hack/filter-crd/main.go

```diff
@@ -69,30 +96,89 @@ func checkChain(d map[interface{}]interface{}, chain []string) {
 		if key, ok := k.(string); ok {
 			chain = append(chain, key)
 
+			// check if keys need to be removed
 			for _, removeKey := range removeKeys {
 				if strings.Join(chain, "/") == removeKey {
 					delete(d, key)
 				}
 			}
 
+			// checks if keys need to be removed when a value is set
+			if value, ok := removeSliceElemForValue[strings.Join(chain, "/")]; ok && value == v.(string) {
+				// TODO
+			}
+
 			if value, ok := v.(map[interface{}]interface{}); ok {
 				checkChain(value, chain)
 			}
+			if value, ok := v.([]interface{}); ok {
+				d[k] = checkSliceChain(value, append(chain, "[]"))
+			}
 			chain = chain[:len(chain)-1] // we're done with this key, remove it from the chain
 		}
 	}
 }
 
+func checkSliceChain(s []interface{}, chain []string) []interface{} {
+	for _, sliceVal := range s {
+		if d, ok := sliceVal.(map[interface{}]interface{}); ok {
+			for k, v := range d {
+				if key, ok := k.(string); ok {
+					chain = append(chain, key)
+
+					// check if keys need to be removed
+					for _, removeKey := range removeKeys {
+						if strings.Join(chain, "/") == removeKey {
+							delete(d, key)
+						}
+					}
+
+					if value, ok := removeSliceElemForValue[strings.Join(chain, "/")]; ok && value == v.(string) {
+						newSlice := []interface{}{}
+
+						for _, sliceVal := range s {
```

Comment:
Hmm.. it's weird that we are iterating through `s` whilst iterating through `s`...

Reply:
Let me move this to a seperate function to make more clear what this does

Reply:
Ah, I think I've just worked it out. It accumulates a new slice to replace the existing one, and does that by searching through all of them and excluding 'itself'